modules/auxiliary/gather/fortios_vpnssl_traversal_creds_leak.rb

```diff
@@ -112,6 +111,11 @@ def parse_config(chunk)
   end
 
   def report_creds(creds)
+    unless framework.database.active
+      print_warning('Tried to save creds but there is no active database that Metasploit is connected to!')
+      return
+    end
```

Comment:
I haven't checked this in detail yet, but shouldn't the credential storing mechanism be noop if a database isn't present, rather than raising an error? 👀

Reply:
Looks like they already have guard clauses in place

https://github.com/rapid7/metasploit-framework/blob/57621d7f5f3afae5a47300402b58f1cb7a257e42/lib/msf/core/auxiliary/report.rb#L36-L52

Was a stacktrace been raised in this scenario? Or is this aimed at being a nicer UX?

Reply:
This was aimed to prevent `myworkspace_id` from being used in the case where the database was not connected.

Reply:
If the database isn't connected `myworkspace_id` should return nil:
https://github.com/rapid7/metasploit-framework/blob/57621d7f5f3afae5a47300402b58f1cb7a257e42/lib/msf/core/auxiliary/report.rb#L90-L96

And it looks like `create_credential_login` ignores its opts if there's no database present 👀

Reply:
Good point, I'll undo this edit then, seeing as it is based on incorrect assumptions. Apologies for that!

Reply:
Removed this code with last rebase, so I'm marking this as resolved.